api/v1/intrusiondetection_types.go

```diff
@@ -27,8 +27,41 @@ type IntrusionDetectionSpec struct {
 	// Only DeepPacketInspection is supported for this spec.
 	// +optional
 	ComponentResources []IntrusionDetectionComponentResource `json:"componentResources,omitempty"`
+
+	// AnomalyDetection provides configuration for running AnomalyDetection Component within
+	// IntrusionDetection
+	// +optional
+	AnomalyDetection AnomalyDetectionSpec `json:"anomalyDetection,omitempty"`
+}
+
+type AnomalyDetectionSpec struct {
+	// StorageType sets the type of storage to use for storing Anomaly Detection Models. By default it will use the ephemeral
+	// emptyDir on the node Anomaly Detection will be deployed to.
+	// default: Ephemeral
+	// +optional
+	// +kubebuilder:validation:Enum=Ephemeral;Persistent
```

Comment:
I believe this validation will mean that an empty string will not be accepted here, which is probably why pointers are preferred (since it won't try to validate a pointer against it?) 

Worth double-checking the behavior of this when StorageType = ""

Reply:
just checked this, was able to provide an empty `anomalydetection:` in the IntrustionDetection CR and it updated to Ephemeral by our `fillDefaults(...)` function

Reply:
Is it possible to have a StorageClassName and still be Ephemeral?

Reply:
https://github.com/tigera/operator/pull/2094/files#diff-ea4aad85809ff62082b52040a934ebbf3953095d563e545bec7daa4d295f2da9R609-R618 our `filldefaults` overrides the type as perssitent if a storageclassname is provided